python_modules/dagster/dagster_tests/definitions_tests/declarative_automation_tests/daemon_tests/test_e2e.py

```diff
@@ -176,15 +176,20 @@ def _execute_ticks(
         )
     )
 
+    backfill_daemon_futures = {}
     list(
         execute_backfill_iteration(
             context,
             get_default_daemon_logger("BackfillDaemon"),
+            threadpool_executor=threadpool_executor,
```

Comment:
Would it make sense to also keep testing the non-threaded route (not just for backfill daemon, but asset and sensor daemons, too?).

Reply:
Yeah, I think we should test both, since the feature will be opt in.

Reply:
this test suite is mostly for auto-materialize i think (this is testing the relatively rare case currently where AMP happens to emit a backfill)

What I would consider the 'real' backfill test suite is here I think: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster_tests/daemon_tests/test_backfill.py#L1-L2791

And here: https://github.com/dagster-io/dagster/blob/0ae36df912c034530d80404ab99e843578687086/python_modules/dagster/dagster_tests/core_tests/execution_tests/test_asset_backfill.py#L1-L1843

Although I don't think we need to include every single test case there or anything, but getting some non-trivial job backfills and asset backfills in there would be a good idea i think (particularly ones that take multiple iterations)

Reply:
> Although I don't think we need to include every single test case there or anything, but getting some non-trivial job backfills and asset backfills in there would be a good idea i think (particularly ones that take multiple iterations)

@gibsondan Can you help point me to some good non-trivial backfills? Would it make sense to test on the `asset_f`/`asset_g` selection?

I added parallel tests for a few of the simpler cases (not 100% sure if I'm using `wait_for_futures` correctly...).

Reply:
> Yeah, I think we should test both, since the feature will be opt in.

Looked into this a bit. In this specific place, all the other daemons are already tested with parallel execution, so maybe it's not so bad to just test the parallel backfill route in this case?

Alternatively, could make the `_execute_tick` helper be parametrized for parallel execution, and test everything both ways. Just thought would check first before going this route, if it made sense.